### PR TITLE
Modified autotuning for flex_attention to pass in (proper) fake inputs for the block sparse entries

### PR DESCRIPTION
**Pull request by Chillee**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #129915
* #129950
* #129846

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/kernel/flex_attention.py [hunk 1]
```diff
@@ -450,6 +456,23 @@ def _get_default_config_bwd(query) -> Tuple[int, int, int, int]:
         return (16, 16, 4, 1)
 
 
+def create_num_blocks_fake(x, num_blocks_for_autotuning) -> torch.Tensor:
+    return torch.full(
+        x.get_size(),
+        int(num_blocks_for_autotuning),
+        dtype=x.get_dtype(),
+        device=x.get_device(),
+    )
```
Comment (yanboliang) [inline torch/_inductor/kernel/flex_attention.py:465]: Does this return a 4D tensor? I think the num blocks tensor should be 3D?

Comment (yanboliang) [reply torch/_inductor/kernel/flex_attention.py:465]: Or it doesn't matter, it's just a fake placeholder?

Comment (yanboliang) [reply torch/_inductor/kernel/flex_attention.py:465]: Addressed offline, my understanding it wrong, the input is the block sparse num blocks, not query.

**Review by yanboliang — APPROVED** [review]

File: torch/_inductor/kernel/flex_attention.py [hunk 2]
```diff
@@ -561,9 +584,21 @@ def flex_attention(*args, **kwargs):
         sparse_kv_num_blocks,
         sparse_kv_indices,
     ] + list(other_buffers)
+    num_blocks_for_autotuning = min(10, sparse_kv_indices.get_size()[-1])
```
Comment (drisspg) [inline torch/_inductor/kernel/flex_attention.py:587]: can you explain this line?

Comment (Chillee) [reply torch/_inductor/kernel/flex_attention.py:587]: oh yeah, I was gonna add a comment.

Comment (Chillee) [reply torch/_inductor/kernel/flex_attention.py]: https://github.com/pytorch/pytorch/pull/129915/files https://github.com/pytorch/pytorch/pull/129915/files#diff-7a9af76999e838f26a7171771bae1dd3243aefeed3827a8412f2f19fbf7b8caaR459

**Review by eellison — APPROVED** [review]

**Pull request closed without merging**